api/src/main/java/org/pmiops/workbench/api/BugReportController.java

```diff
@@ -44,7 +72,32 @@
       msg.addRecipient(Message.RecipientType.TO, new InternetAddress(
           workbenchConfig.admin.supportGroup, "AofU Workbench Engineers"));
       msg.setSubject("[AofU Bug Report]: " + bugReport.getShortDescription());
-      msg.setText(bugReport.getReproSteps());
+
+      Multipart multipart = new MimeMultipart();
+      MimeBodyPart textPart = new MimeBodyPart();
+      textPart.setText(bugReport.getReproSteps());
+      multipart.addBodyPart(textPart);
+
+      // If requested, try to pull logs from the notebook cluster using the researcher's creds. Some
+      // or all of these log files might be missing, or the cluster may not even exist, so ignore
+      // failures here.
+      if (Optional.ofNullable(bugReport.getIncludeNotebookLogs()).orElse(false) &&
+          BillingProjectStatus.READY.equals(user.getFreeTierBillingProjectStatus())) {
+        for (String fileName : BugReportController.notebookLogFiles) {
+          try {
+            String logContent = jupyterApi.getRootContents(
+                user.getFreeTierBillingProjectName(), NotebooksService.DEFAULT_CLUSTER_NAME,
+                fileName, "file", "text", /* content */ 1).getContent();
+            MimeBodyPart attachPart = new MimeBodyPart();
+            attachPart.setDataHandler(new DataHandler(logContent, "text/plain"));
+            attachPart.setFileName(fileName);
+            multipart.addBodyPart(attachPart);
+          } catch (ApiException e) {
+            log.info(String.format("failed to retrieve notebook log '%s', continuing", fileName));
```

Comment:
Should this be a slightly higher log level, warn maybe? I'm fine with leaving it as info if we expect that might be too noisy.

Reply:
One more thought. Any chance of an NPE from `logContent`?

Reply:
On log level, I think info is OK here as it's expected that these files may not exist, or the cluster may not even exist.

On the NPE - in theory that shouldn't happen. In practice I'm not sure and we're in the hands of the Jupyter contents API; added guarding against this.